config/config.go

```diff
@@ -398,6 +398,16 @@ func (c *ScrapeConfig) UnmarshalYAML(unmarshal func(interface{}) error) error {
 	if c.BasicAuth != nil && (len(c.BearerToken) > 0 || len(c.BearerTokenFile) > 0) {
 		return fmt.Errorf("at most one of basic_auth, bearer_token & bearer_token_file must be configured")
 	}
+	// Check for users putting URLs in target groups.
+	if len(c.RelabelConfigs) == 0 {
+		for _, tg := range c.TargetGroups {
+			for _, t := range tg.Targets {
+				if strings.Contains(string(t[model.AddressLabel]), "/") {
```

Comment:
Our expectation of a valid address is even more well defined, isn't it? Like `hostname:port` (but I'd need to check). Maybe add a comment to improve that here.

It would also make sense to extract this condition into a function and reuse it in the target manager.